rpcs3/Emu/SysCalls/Modules/cellSysutil.cpp

```diff
@@ -224,12 +223,17 @@ s32 cellSysCacheMount(vm::ptr<CellSysCacheParam> param)
 {
 	cellSysutil.Warning("cellSysCacheMount(param=*0x%x)", param);
 
-	//TODO: implement
+	// TODO: implement
 	char id[CELL_SYSCACHE_ID_SIZE];
 	strncpy(id, param->cacheId, CELL_SYSCACHE_ID_SIZE);
 	strncpy(param->getCachePath, ("/dev_hdd1/cache/" + std::string(id) + "/").c_str(), CELL_SYSCACHE_PATH_MAX);
 	param->getCachePath[CELL_SYSCACHE_PATH_MAX - 1] = '\0';
-	Emu.GetVFS().CreateDir(std::string(param->getCachePath));
+	std::string directory = param->getCachePath;
+
+	if (!Emu.GetVFS().ExistsDir(directory))
```

Comment:
There is no point in calling two syscalls when you can just try to create a directory and analyse the result of makedir or something. Checking its existence is also slower, plus it's non-atomic. For example, if the directory got created by another thread just after the first thread had checked dir's existence, it'll fail to create directory anyway, thinking that it doesn't exist. And getting rid of errors in log without fixing the real problems is usually just silly.